### PR TITLE
Feature/datasource ssh keys

### PR DESCRIPTION
**Pull request by riethm**
- Added a datasource `softlayer_ssh_key`, which can be used to import ssh keys into the local state, using `label` as the import key
- Removed the `ssh_key_labels` property from the softlayer_virtual_guest resource schema
- Renamed `name` to `label` in the softlayer_ssh_key resource schema, to reflect the name of the field as is commonly visible to users in the portal or CLI (`name` is an API field)

### CODE REVIEW

File: docs/datasources/softlayer_ssh_key.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# `softlayer_ssh_key`
+
+Use this data source to import the details of an *existing* SSH key as a read-only data source.
```
Comment (renier) [inline docs/datasources/softlayer_ssh_key.md]: I would add that you can use this to get the SSH key id that you could use in other resources like virtual_guest.

File: docs/datasources/softlayer_ssh_key.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+The fields of the data source can then be referenced by other resources within the same configuration using
+interpolation syntax. For example, when specifying SSH keys in a softlayer_virtual_guest resource configuration,
+the numeric "IDs" are often unknown. Using the above data source as an example, it would be possible to
+reference the `id` property in a softlayer_virtual_guest resource:
+
+```hcl
+resource "softlayer_virtual_guest" "vm1" {
+    ...
+    ssh_keys = ["${data.softlayer_ssh_key.public_key.id}"]
```
Comment (renier) [inline docs/datasources/softlayer_ssh_key.md:21]: will this result in a string or an int?

File: docs/datasources/softlayer_ssh_key.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+```
+
+## Argument Reference
+
+* `label` - (Required) The label of the SSH key, as it was defined in SoftLayer
+* `most_recent` - (Optional) If more than SSH key matches the label, use the most recent key
+
+NOTE: If more or less than a single match is returned by the search, Terraform will fail.
+Ensure that your search is specific enough to return a single SSH key only,
```
Comment (renier) [inline docs/datasources/softlayer_ssh_key.md]: Ensure that your label...

File: docs/datasources/softlayer_ssh_key.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+or use most_recent to choose the most recent one.
```
Comment (renier) [inline docs/datasources/softlayer_ssh_key.md]: or use `*most_resent*`...

File: docs/softlayer_virtual_guest.md [hunk 5]
```diff
@@ -118,10 +118,9 @@ The following arguments are supported:
     * *Default*: nil
     * *Optional*
     * *Conflicts with* `ssh_key_labels`
```
Comment (renier) [inline docs/softlayer_virtual_guest.md]: Need to remove this line, too.

File: docs/datasources/softlayer_ssh_key.md (hunk 2 above)
Comment (riethm) [reply docs/datasources/softlayer_ssh_key.md:21]: Good question - I have never been able to get it to work with quotes omitted.  Here's what I get when I use this (quoted) syntax:

```
ssh_keys.#:               "1"
ssh_keys.0:               "383111"
```

So clearly, string.  But `ssh_keys = [383111]` yields the exact same result.  Maybe Terraform stores everything as `interface{}`, displays as strings, and type asserts as needed.

File: softlayer/resource_softlayer_ssh_key.go [hunk 6]
```diff
@@ -28,11 +28,17 @@ func resourceSoftLayerSSHKey() *schema.Resource {
...
 				Type:     schema.TypeString,
 				Required: true,
 			},
 
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Removed:  "Use 'label' instead",
+			},
+
```
Comment (renier) [inline softlayer/resource_softlayer_ssh_key.go]: I would not bother with deprecations... Up to you. and it's nice that they have _Removed_ for these situations, but I don't think we are there yet to need it.

File: softlayer/resource_softlayer_virtual_guest.go [hunk 7]
```diff
@@ -187,17 +187,16 @@ func resourceSoftLayerVirtualGuest() *schema.Resource {
...
 			"ssh_key_labels": {
-				Type:          schema.TypeList,
-				Optional:      true,
-				Elem:          &schema.Schema{Type: schema.TypeString},
-				ConflictsWith: []string{"ssh_keys"},
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Removed:  fmt.Sprintf("Use the interpolation method described at %s", datasourceSshKeyDoc),
 			},
```
Comment (renier) [inline softlayer/resource_softlayer_virtual_guest.go]: This has been there for such a short time, I'd just remove.

**Review by renier — COMMENTED** [review]
I think we need to evolve our docs directory. Looking at the terraform project, they have the provider documentation, then two subdirectories. One for resources, another for data sources. I would like to see us emulate that.

File: docs/datasources/softlayer_ssh_key.md (hunk 2 above)
Comment (renier) [reply docs/datasources/softlayer_ssh_key.md:21]: ugh, and `ssh_keys = [${data.softlayer_ssh_key.public_key.id}]` would not work then? That means I made a mistake in some of the doc updates.

File: docs/datasources/softlayer_ssh_key.md (hunk 3 above)
Comment (riethm) [reply docs/datasources/softlayer_ssh_key.md]: Fixed

File: docs/datasources/softlayer_ssh_key.md (hunk 4 above)
Comment (riethm) [reply docs/datasources/softlayer_ssh_key.md]: Fixed

File: docs/softlayer_virtual_guest.md (hunk 5 above)
Comment (riethm) [reply docs/softlayer_virtual_guest.md]: Fixed

File: softlayer/resource_softlayer_ssh_key.go (hunk 6 above)
Comment (riethm) [reply softlayer/resource_softlayer_ssh_key.go]: Removed `Removed`

File: softlayer/resource_softlayer_virtual_guest.go (hunk 7 above)
Comment (riethm) [reply softlayer/resource_softlayer_virtual_guest.go]: Removed `Removed`

**Pull request merged**